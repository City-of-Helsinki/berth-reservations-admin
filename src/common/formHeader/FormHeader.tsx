import React from 'react';
import classNames from 'classnames';
import { useTranslation } from 'react-i18next';
import Text from '../text/Text';
import styles from './formHeader.module.scss';

interface Props {
  title: string;
  className?: string;
  upperCase?: boolean;
  isSubmitting?: boolean;
  onDeleteText?: string;
  onDelete?(): void;
}

const FormHeader = ({ title, className, isSubmitting, upperCase, onDelete, onDeleteText }: Props) => {
  const { t } = useTranslation();
  return (
    <div className={classNames(styles.heading, className)}>
      <Text as="h4" color="brand" className={classNames({ [styles.upperCase]: upperCase })}>
        {title}
      </Text>
      {onDelete && (
        <button disabled={isSubmitting} onClick={() => onDelete()} type="button">
          <Text color="critical">{onDeleteText ?? t('forms.common.delete')}</Text>
        </button>
      )}
    </div>
  );
};

export default FormHeader;
