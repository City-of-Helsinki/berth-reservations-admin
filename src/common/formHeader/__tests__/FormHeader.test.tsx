import React from 'react';
import { shallow } from 'enzyme';

import FormHeader from '../FormHeader';

const mockProps = {
  title: 'Test',
};

describe('FormHeader', () => {
  const getWrapper = (props = {}) => shallow(<FormHeader {...mockProps} {...props} />);

  it('renders normally with minimum props', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders normally with all props', () => {
    const wrapper = getWrapper({
      isSubmitting: true,
      onDeleteText: 'Delete this',
      onDelete: jest.fn(),
    });

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('onDelete should be called if the button is clicked', () => {
    const onDelete = jest.fn();
    const wrapper = getWrapper({
      onDelete: onDelete,
    });

    wrapper.find('button').simulate('click');

    expect(onDelete).toHaveBeenCalled();
  });
});
