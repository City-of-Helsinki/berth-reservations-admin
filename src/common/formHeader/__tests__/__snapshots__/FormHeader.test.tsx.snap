// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormHeader renders normally with all props 1`] = `
<div
  class="heading"
>
  <h4
    class="text brand h4"
  >
    Test
  </h4>
  <button
    disabled=""
    type="button"
  >
    <span
      class="text critical span"
    >
      Delete this
    </span>
  </button>
</div>
`;

exports[`FormHeader renders normally with minimum props 1`] = `
<div
  class="heading"
>
  <h4
    class="text brand h4"
  >
    Test
  </h4>
</div>
`;
